crates/nargo/src/cli/test_cmd.rs

```diff
@@ -0,0 +1,86 @@
+use std::io::Write;
+
+use acvm::ProofSystemCompiler;
+use clap::ArgMatches;
+use noirc_driver::{CompiledProgram, Driver};
+use noirc_frontend::node_interner::FuncId;
+use termcolor::{Color, ColorChoice, ColorSpec, StandardStream, WriteColor};
+
+use crate::{errors::CliError, resolver::Resolver};
+
+use super::add_std_lib;
+
+pub(crate) fn run(args: ArgMatches) -> Result<(), CliError> {
+    let args = args.subcommand_matches("test").unwrap();
+    let test_name = args.value_of("test_name").unwrap_or("");
+    let allow_warnings = args.is_present("allow-warnings");
+    run_tests(test_name, allow_warnings)
+}
+
+fn run_tests(test_name: &str, allow_warnings: bool) -> Result<(), CliError> {
+    let backend = crate::backends::ConcreteBackend;
+
+    let package_dir = std::env::current_dir().unwrap();
+    let mut driver = Resolver::resolve_root_config(&package_dir, backend.np_language())?;
+    add_std_lib(&mut driver);
+
+    if driver.check_crate(allow_warnings).is_err() {
+        std::process::exit(1);
+    }
+
+    let test_functions = driver.get_all_test_functions_in_crate_matching(test_name);
+    println!("Running {} test functions...", test_functions.len());
+    let mut failing = 0;
+
+    let writer = StandardStream::stderr(ColorChoice::Always);
+    let mut writer = writer.lock();
+
+    for test_function in test_functions {
+        let test_name = driver.function_name(test_function);
+        write!(writer, "Testing {test_name}... ").expect("Failed to write to stdout");
+        writer.flush().ok();
+
+        match prove(test_name, test_function, &driver, allow_warnings) {
+            Ok(_) => {
+                writer.set_color(ColorSpec::new().set_fg(Some(Color::Green))).ok();
+                writeln!(writer, "ok").ok();
+            }
+            Err(_) => {
+                // An error likely was printed in the meantime, so this is the start of a newline.
+                // Add an extra newline as well to help separate failing tests.
+                write!(writer, "{test_name} ").ok();
+                writer.set_color(ColorSpec::new().set_fg(Some(Color::Red))).ok();
+                println!("failed\n");
+                failing += 1;
+            }
+        }
+        writer.reset().ok();
+    }
+
+    if failing == 0 {
+        writer.set_color(ColorSpec::new().set_fg(Some(Color::Green))).unwrap();
+        writeln!(writer, "All tests passed").ok();
+    } else {
+        let plural = if failing == 1 { "" } else { "s" };
+        writer.set_color(ColorSpec::new().set_fg(Some(Color::Red))).unwrap();
+        writeln!(writer, "{failing} test{plural} failed").ok();
+        std::process::exit(1);
+    }
+
+    writer.reset().ok();
+    Ok(())
+}
+
+fn prove(
```

Comment:
This is /bikeshedding but is `prove` the best name here as we are not proving anything. `execute`, `run`?

Reply:
Those would be better names, this name was from before e1e2d1e93d78e0b175d146ca827ca23da38588f4 where it was indeed running the whole prover